pkg/operator/ceph/object/spec.go

```diff
@@ -338,6 +335,9 @@ func (c *clusterConfig) generateService(cephObjectStore *cephv1.CephObjectStore)
 		svc.Spec = v1.ServiceSpec{
 			Selector: getLabels(cephObjectStore.Name, cephObjectStore.Namespace, false),
 		}
+		if c.store.Spec.Gateway.SecurePort != 0 && c.store.Spec.GetServiceServingCert() != "" {
+			c.store.Spec.Gateway.Service.Annotations.ApplyToObjectMeta(&svc.ObjectMeta)
```

Comment:
Why don't we just *always* apply annotations instead of only if the OpenShift annotation we look for is set?

Reply:
Extending my thought a little bit... what if instead of hard-coding onto an openshift feature present in an annotation, we just had an option that told Rook not to verify the cert from a secret? Like if a user has another way to provide certs for the RGW without passing them as a secret, even if that way isn't by setting an annotation.

Maybe something like `Spec.Gateway.DoNotUseTLSCertificate` (default false) but users still have to specify `Spec.Gateway.SecurePort`

Reply:
Yeah, I agree that we need to copy the entire annotation than this specific one(the earlier patch set I was using annotation from `GatewaySpec`, not this one). IMO `Spec.Gateway.DoNotUseTLSCertificate` may give the wrong message in either rook automatically create TLS certs for them or allows them access `https` without certs. So I hope to keep this way is more clear.